includes/Classifai/Providers/Azure/TextToSpeech.php

```diff
@@ -0,0 +1,673 @@
+					'api_key' => $new_settings['credentials']['api_key'],
+				),
+				true
+			);
+
+			if ( is_wp_error( $is_connected ) ) {
+				$new_settings['authenticated'] = false;
+			} else {
+				$new_settings['authenticated'] = true;
+			}
+		} else {
+			$new_settings['credentials']['url']           = '';
+			$new_settings['credentials']['api_key']       = '';
+			$new_settings['authenticated'] = false;
+
+			add_settings_error(
+				$this->get_option_name(),
+				'classifai-azure-text-to-speech-auth-empty',
+				esc_html__( 'One or more credentials required to connect to the Azure Text to Speech service is empty.', 'classifai' ),
+				'error'
+			);
+		}
+
+		// Sanitize the post type checkboxes
+		$post_types = get_post_types_for_language_settings();
+
+		foreach ( $post_types as $post_type ) {
+			if ( isset( $settings['post_types'][ $post_type->name ] ) ) {
+				$new_settings['post_types'][ $post_type->name ] = $settings['post_types'][ $post_type->name ];
+			} else {
+				$new_settings['post_types'][ $post_type->name ] = null;
+			}
+		}
+
+		if ( isset( $settings['voice'] ) && ! empty( $settings['voice'] ) ) {
+			$new_settings['voice'] = sanitize_text_field( $settings['voice'] );
+		} else {
+			$new_settings['voice'] = '';
+		}
+
+		return $new_settings;
+	}
+
+	/**
+	 * Connects to Azure's Text to Speech service.
+	 *
+	 * @param array   $args             Overridable args.
+	 * @param boolean $ignore_transient Boolean to ignore transient. Used to reconnect to the service.
+	 * @return boolean|null
+	 */
+	public function connect_to_service( array $args = array(), $ignore_transient = false ) {
+		$creditials = $this->get_settings( 'credentials' );
+
+		$default = array(
+			'url'     => isset( $creditials['url'] ) ? $creditials['url'] : '',
+			'api_key' => isset( $creditials['api_key'] ) ? $creditials['api_key'] : '',
+		);
+
+		$default = array_merge( $default, $args );
+
+		$voices_transient = get_transient( 'classifai-azure-speech-to-text-voices' );
+
+		// Return if transient exists.
+		if ( false !== $voices_transient && $ignore_transient ) {
+			return null;
+		}
+
+		// Return if credentials don't exist.
+		if ( empty( $default['url'] ) || empty( $default['api_key'] ) ) {
+			return false;
+		}
+
+		// Create request arguments.
+		$request_params = array(
+			'headers' => array(
+				'Ocp-Apim-Subscription-Key' => $default['api_key'],
+				'Content-Type'              => 'application/json',
+			),
+		);
+
+		// Create request URL.
+		$request_url = sprintf(
+			'%1$scognitiveservices/voices/list',
+			$default['url']
+		);
+
+		if ( function_exists( 'vip_safe_wp_remote_get' ) ) {
+			$response = vip_safe_wp_remote_get(
+				$request_url,
+				'',
+				3,
+				1,
+				20,
+				$request_params
+			);
+		} else {
+			// phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.wp_remote_get_wp_remote_get -- use of `vip_safe_wp_remote_get` is done when available.
+			$response = wp_remote_get(
```

Comment:
Do we need to set a higher timeout value here? The default is 5 seconds, not sure if we ever come close to exceeding that.

Reply:
@dkotter it fails due to timeout while testing with large text content, that's why for safety I increased the value.